src/core/biblio.js

```diff
@@ -137,19 +93,78 @@ export async function run(conf) {
     idbRefs.push(...neededRefs.map(id => ({ id, data: null })));
     console.warn(err);
   }
-  const split = { hasData: [], noData: [] };
-  idbRefs.forEach(ref => {
-    (ref.data ? split.hasData : split.noData).push(ref);
-  });
-  split.hasData.forEach(ref => {
-    biblio[ref.id] = ref.data;
-  });
-  const externalRefs = split.noData.map(item => item.id);
-  if (externalRefs.length) {
-    // Going to the network for refs we don't have
-    const data = await updateFromNetwork(externalRefs, { forceUpdate: true });
-    Object.assign(biblio, data);
+
+  return idbRefs;
+}
+
+export class Class {
+  /** @param {Conf} conf */
+  constructor(conf) {
+    this.conf = conf;
+  }
+
+  /**
+   * Normative references take precedence over informative ones,
+   * so any duplicates ones are removed from the informative set.
+   */
+  normalizeReferences() {
```

Comment:
A separate thought: I don't quite like how this affects other modules... 🤔

Reply:
Another separate thought: `normativeReferences` should not exist on `conf` (keep conf for user defined cobfig only?). We should rather `export const normativeReferences = new Set()` from the module where it's created

Reply:
agree... of some kind of promise getter.